File: db2/rhino/update/db2_update.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ----------------------------------------------------------------------
/*
	db2_update.js

				Jun/02/2011

*/
// ----------------------------------------------------------------------
importPackage (java.sql);
importPackage (java.lang);

load ("/var/www/data_base/common/rhino_common/sql_manipulate.js");
load ("/var/www/data_base/common/rhino_common/text_manipulate.js");
// ----------------------------------------------------------------------
print ("*** 開始 ***");

var id_in=arguments[0];
var population_in=arguments[1];

print ("\tid_in = " + id_in);
print ("\tpopulation_in = " + population_in);

Class.forName("COM.ibm.db2.jdbc.app.DB2Driver").newInstance();

var url = "jdbc:db2:cities";

var user = "scott";
var pass = "tiger";

var conn = DriverManager.getConnection (url,user,pass);

update_proc	(conn,id_in,population_in);

disp_proc	(conn);

conn.close ();

print ("*** 終了 ***");

// ----------------------------------------------------------------------

File: db2/jruby/update/db2_update.rb
#! /usr/bin/jruby
# -*- encoding: utf-8 -*-
#
#	db2_update.rb
#
#					Jun/09/2011
#
#
# ---------------------------------------------------------------------
include Java
import java.sql.DriverManager
import java.lang.System
#
require 'date'
#
load '/var/www/data_base/common/jruby_common/jruby_sql_manipulate.rb'
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
id_in = ARGV[0]
population_in = ARGV[1].to_i
#
puts id_in,population_in
#
Java::JavaClass.for_name("COM.ibm.db2.jdbc.app.DB2Driver")

url = "jdbc:db2:cities"

user = "scott"
pass = "tiger"

conn = DriverManager.getConnection(url,user,pass)

sql_update_proc(conn,id_in,population_in)

sql_read_proc(conn)

conn.close()

puts	"*** 終了 ***"

# ---------------------------------------------------------------------
